utils/extractor/src/index.ts

```diff
@@ -293,3 +293,53 @@ export type SheetCapture = {
   data: Flatfile.RecordData[]
   metadata?: { rowHeaders: number[] }
 }
+
+export function parseSheet(jsonArray: any[]): SheetCapture {
+  try {
+    // Ensure all items are objects
+    const filteredResults = jsonArray.filter(
+      (item) => typeof item === 'object' && item !== null
+    )
+
+    if (filteredResults.length === 0) {
+      return {} as SheetCapture
+    }
+
+    // Custom flatten function
+    const flattenObject = (obj: any, parent: string = '', res: any = {}) => {
```

Comment:
Suggestion from Patrick to not mutate arguments and instead copy values and return the changed result https://github.com/FlatFilers/flatfile-plugins/pull/693/files#r1826028605

I didn't want to change this refactored code too much as the goal was simply to move it to make it available to both JSON extractors. However, recursive functions typically tend to leverage this mutation pattern, and it's almost idiomatic. I'll add a comment that notes as much

Note that using a non mutating recursive approach adds significantly complexity to merging the final results, likely making them much slower for large datasets